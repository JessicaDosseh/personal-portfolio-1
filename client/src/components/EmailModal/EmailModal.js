import React from "react";
import PropTypes from "prop-types";
import styled, { css, keyframes } from "styled-components";

const buttonHover = keyframes`
  0% {
    opacity: 1;
  }
  100% {
    opacity: 0.8;
  }
`;

const buttonHoverDark = keyframes`
  0% {
    opacity: 1;
  }
  100% {
    opacity: 0.6;
  }
`;

const Modal = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  padding: 0 5%;
  background: white;
  width: 100%;
  min-height: 100vh;
  z-index: 50;

  ${props =>
    props.darkmode &&
    css`
      background: #2f3234;
    `}
  div {
    height: 150px;
    max-width: 400px;
    width: 100%;
    margin: 0 auto;
    z-index: 51;
    margin-top: 30vh;
    display: flex;
    flex-direction: column;

    p {
      width: 100%;
      text-align: center;
      color: black;
      margin-bottom: 35px;

      ${props =>
        props.darkmode &&
        css`
          color: white;
        `}
    }

    button {
      width: 100%;
      height: 45px;
      border: 0;
      border-radius: 5px;
      -moz-border-radius: 5px;
      -webkit-border-radius: 5px;
      background: #ad91ed;
      /* #b6465f */
      color: white;
      font-size: 1.6rem;
      text-transform: capitalize;
      cursor: pointer;

      ${props =>
        props.darkmode &&
        css`
          background: black;
        `}

      &:hover {
        animation: ${buttonHover} 0.2s forwards;

        ${props =>
          props.darkmode &&
          css`
            animation: ${buttonHoverDark} 0.2s forwards;
          `}
      }
    }
  }
`;

const EmailModal = props => {
  const clickBtn = e => {
    e.preventDefault();
    props.setEmailValues({
      success: null,
      displayModal: false
    });
    props.history.push("/");
  };
  return (
    <Modal darkmode={props.darkmode}>
      <div>
        <p>
          {props.emailValues.success ? "Message sent" : "Error - Try again"}
        </p>
        <button onClick={clickBtn}>Okay</button>
      </div>
    </Modal>
  );
};

EmailModal.propTypes = {
  darkmode: PropTypes.bool
};

export default EmailModal;
