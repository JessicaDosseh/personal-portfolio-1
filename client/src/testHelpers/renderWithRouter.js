import React from "react";
import { Router } from "react-router-dom";
import { render } from "react-testing-library";
import { createMemoryHistory } from "history";
import { StateProvider } from "react-conflux";
import { darkContext } from "../store/contexts";
import { globalReducer } from "../store/reducers/globalReducer";

const renderWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(
      <StateProvider reducer={globalReducer} stateContext={darkContext}>
        <Router history={history}>{ui}</Router>
      </StateProvider>
    ),
    history
  };
};

export default renderWithRouter;
