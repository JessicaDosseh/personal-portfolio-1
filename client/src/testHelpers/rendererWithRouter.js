import React from "react";
import { Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";
import { StateProvider } from "react-conflux";
import { darkContext } from "../store/contexts";
import { globalReducer } from "../store/reducers/globalReducer";

const rendererWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(
      <StateProvider reducer={globalReducer} stateContext={darkContext}>
        <Router history={history}>{ui}</Router>
      </StateProvider>
    ),
    history
  };
};

export default rendererWithRouter;
