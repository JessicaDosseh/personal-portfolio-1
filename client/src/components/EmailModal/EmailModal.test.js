import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { renderWithRouter, rendererWithRouter } from "../../testHelpers";

import EmailModal from "./EmailModal";

describe("<EmailModal />", () => {
  it("should render the EmailModal component without crashing", () => {
    const helpers = renderWithRouter(
      <EmailModal darkmode={false} emailValues={{ success: true }} />
    );
  });

  it("should match the snapshot of EmailModal", () => {
    const tree = rendererWithRouter(
      <EmailModal darkmode={false} emailValues={{ success: true }} />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
