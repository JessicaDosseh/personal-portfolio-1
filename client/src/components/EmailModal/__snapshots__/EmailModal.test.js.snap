// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailModal /> should match the snapshot of EmailModal 1`] = `
<div
  className="sc-bdVaJa lnxrCL"
>
  <div>
    <p>
      Message sent
    </p>
    <button
      onClick={[Function]}
    >
      Okay
    </button>
  </div>
</div>
`;
